### ISSUE TITLE
Add notification Permissions

### ISSUE THREAD

**Issue by Vanethos** [FogosPT/fogosmobile#26]
Add notification permissions (you can use [this package](https://pub.dev/packages/notification_permissions))

The current requirements are:
- Ask 1 time to the user when he first installs the app. You can share a value in the shared preferences to state that the user has seen the message
- Every time the user opens the `Notifications` screen, check if he has permissions, he he does not have permissions, show him the message.

**Comment by miguelpruivo** [comment]
Is anyone currently working on this?

**Comment by Vanethos** [comment]
I think @Lima21 is working on this but he stumbled on some issues integrating the library

Did you solve the issue @Lima21 ?

**Comment by Lima21** [comment]
I don't have time to solve the issue however I talked with @0xSmiley and I think he is working on it.

**Comment by Vanethos** [comment]
Could you un-assign yourself from the issue @Lima21 ? This way we can know who is working on the issue.

And thank you very much for your time!

**Comment by miguelpruivo** [comment]
Also @Vanethos, there should be no reason to save the permission in the local preferences has the system is aware of that (you can always check in the app permission settings). However, it should make sure that the user has or not permission (checking in runtime) to access the demanding feature, and show a message if needed.

**Comment by miguelpruivo** [comment]
Also @Vanethos, IMO there should be no reason to save the permission in the local preferences has the system is aware of that (you can always check in the app permission settings). However, it should make sure that the user has or not permission (checking in runtime) to access the demanding feature, and show a message if needed.

**Comment by Vanethos** [comment]
@miguelfmp it is not if the permission is accepted or not, but if the user has seen the permission dialog before or not.

Else, each time the app launches the user will see that dialog, and he may not want to see that.

**Comment by miguelpruivo** [comment]
@Vanethos you’ve @ the wrong person 😅. Anyway, I’m referring to the system permission dialog, thus, that one will only show once (or until the user accept the permissions). 

If this is related to a custom dialog explaining why we are requiring the notifications, then that’s another use case, and if so, what you’re saying may make sense.

**Comment by Vanethos** [comment]
Yes, you're right, I'm always thinking of the custom dialog

And sorry, too many Miguels here ahahahah

@0xSmiley how are we on this issue?

**Comment by 0xSmiley** [comment]
Hey, sorry for the delayed response.
Currently still trying to implement the library in the project. I'm having some difficulties since when I compile the project theres a missing plugin error.

**Comment by miguelpruivo** [comment]
@0xSmiley feel free to ask for some help if you feel that you’re struggling with it. 

What lib are you trying to use?

**Comment by 0xSmiley** [comment]
I'm using the notification_permissions 0.3.0. After doing flutter get and importing the path I get a 
```fatal error: 'notification_permissions/notification_permissions-Swift.h' file not found```
I have checked the contents of notification_permissions version 0.3.0 and theres is not a file named notification_permissions-Swift.h also couldn't find any information related to the problem online. Any ideia why this is happening?

**Comment by miguelpruivo** [comment]
Looks like you’re trying to add a swift library into the project. Swift libraries need to make sure you’re using some flags in the Pod file and can result in some issues when mixed with Obj-C ones (which I believe are the majority). 

Obj-C libraries are static libs and swift can be dynamic or static. The problem is that when you use obj-c libs, you need to make sure you’re using static swift libs (thus, adding `use_frameworks!` flag in the Pod file). But, some swift libs are not ready to be static compiled and then you start running into issues. 

TL;DR: For now, I’d stay away of Swift libs.

**Comment by 0xSmiley** [comment]
Thank you so much @miguelpruivo for the great response!! I will look into the permission_handler package :)

**Comment by Vanethos** [comment]
@miguelpruivo one question, does that package contain the notification permissions request? 

I remember that when I tried to use it in the past I had some issues, hence creating the new one.

Will work on adding Objective-C to the library asap

**Comment by miguelpruivo** [comment]
@Vanethos, good point. I know it’s very complete as for the permissions in general, but I’m not sure about the notifications one. By giving a quick glance at the docs it looks like it doesn’t. (Somehow I misinterpreted location permissions instead of notifications).

@0xSmiley we’ll probably have to look for another package as that doesn’t offers what we need.

**Comment by Vanethos** [comment]
Isn't it best to try for the short term to integrate the swift library?

**Comment by miguelpruivo** [comment]
@Vanethos if it is able to run statically, sure. I’m wondering if not.

**Comment by Vanethos** [comment]
This afternoon I'll try to incorporate the library and create a commit with it

**Comment by Vanethos** [comment]
@0xSmiley the package has been added in a new branch, FGPT-26.

Please clone that branch and then you can resume your work :D
